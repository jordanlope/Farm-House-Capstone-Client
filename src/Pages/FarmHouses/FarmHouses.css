ul.FarmList {
    margin: 0px;
    padding: 0px;
}

.FarmList li a{
    font-weight: bold;
    background-color: white;
    border: none;
    color: black;
    padding: 5px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 20px;
    width: 100%;
    margin: 4px 2px;
    cursor: pointer;
}

.FarmList li a:hover{
    font-weight: bold;
    background-color: rgba(36, 35, 35, 0.411);
    border: none;
    color: black;
    padding: 5px;
    text-align: center;
    text-decoration: none;
    display: inline-block;
    font-size: 20px;
    width: 100%;
    margin: 4px 2px;
    cursor: pointer;
}

img.MainImage {
    max-width: 100%;
    margin: 0px;
}

h1.title {
    text-shadow: 1px 2px gainsboro;
    /* background-color:lightslategray; */
    text-align: center;
    margin: -3px;
    padding: 20px;
}