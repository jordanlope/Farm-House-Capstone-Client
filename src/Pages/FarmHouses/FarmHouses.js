import React from 'react';
import './FarmHouses.css';
import { Link } from 'react-router-dom';
import FarmHousesService from '../../services/farmHouses-api-service'
import FarmHouseListContext from '../../contexts/FarmHouseListContext'

export default class FarmHouses extends React.Component {
  static contextType = FarmHouseListContext

  componentDidMount() {
    this.context.clearError()
    FarmHousesService.getFarmHouses()
      .then(this.context.setFarmHouseList)
      .catch(this.context.setError)
  }

  render() {
    const { farmHouseList } = this.context ? this.context : []

    const { error } = this.context

    return (
      <>
        <h1>Farm Houses:</h1>
        <ul className='FarmList'>
          {
            farmHouseList.map(house =>
              <li key={house.id}>
                <Link to={`/home/${house.id}`}>
                  
                  <h2>{house.address}</h2>
                  
                  <p>{house.description}</p>
                </Link>
              </li>
          )}
        </ul>
      </>
    );
  }
}