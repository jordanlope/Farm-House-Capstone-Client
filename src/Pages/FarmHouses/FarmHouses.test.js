import React from 'react'
import ReactDOM from 'react-dom'
import FarmHouses from './FarmHouses'
import renderer from 'react-test-renderer'

describe('FarmHouses component', () => {
    // Smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div')
        ReactDOM.render(<FarmHouses />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    //Snapshot test
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<FarmHouses />)
            .toJSON()
        expect(tree).toMatchSnapshot();
    })
})