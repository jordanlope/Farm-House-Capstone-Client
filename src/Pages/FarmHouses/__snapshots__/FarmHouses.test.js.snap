// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FarmHouses component renders the UI as expected 1`] = `
Array [
  <h1>
    Farm Houses:
  </h1>,
  <ul
    className="FarmList"
  />,
]
`;
